ksql-common/src/main/java/io/confluent/ksql/util/KsqlConfig.java

```diff
@@ -196,10 +196,22 @@
           + "whether the Kafka cluster supports the required API, and enables the validator if "
           + "it does.";
 
-  public static final String KSQL_PULL_QUERIES_ENABLE_CONFIG = "ksql.pull.queries.enable";
-  public static final String KSQL_PULL_QUERIES_ENABLE_DOC =
+  public static final String KSQL_QUERY_PULL_ENABLE_CONFIG = "ksql.query.pull.enable";
+  public static final String KSQL_QUERY_PULL_ENABLE_DOC =
       "Config to enable or disable transient pull queries on a specific KSQL server.";
-  public static final boolean KSQL_PULL_QUERIES_ENABLE_DEFAULT = true;
+  public static final boolean KSQL_QUERY_PULL_ENABLE_DEFAULT = true;
+
+  public static final String KSQL_QUERY_PULL_OWNER_TIMEOUT_MS_CONFIG =
+      "ksql.query.pull.owner.timeout.ms";
+  public static final Long KSQL_QUERY_PULL_OWNER_TIMEOUT_MS_DEFAULT = 30000L;
+  public static final String KSQL_QUERY_PULL_OWNER_TIMEOUT_MS_DOC = "Timeout in milliseconds "
+      + "when waiting for the lookup of the owner of a row key";
+
+  public static final String KSQL_STREAMSTORE_REBALANCING_TIMEOUT_MS_CONFIG =
+      "ksql.streamsstore.rebalancing.timeout.ms";
+  public static final Long KSQL_STREAMSTORE_REBALANCING_TIMEOUT_MS_DEFAULT = 10000L;
```

Comment:
Note:  We might not need this param eventually, once KIP-535 on Kafka is implemented and we can query even in rebalancing state. Guess we can back it out later. Good with that.

Reply:
Good to know.  I can stick a TODO there, if that's helpful.

Reply:
I think raising a Github issue (with reference to the KIP) would be better. There was a discussion a while back before you joined about standardizing code standards and we're trying to avoid TODO's 
https://github.com/confluentinc/ksql/pull/3642#discussion_r339044928

Reply:
https://github.com/confluentinc/ksql/issues/3869